### ISSUE TITLE
Cannot change size to an independent disk after attaching it to a VM

### ISSUE THREAD

**Issue by ejgarcia** [vmware/terraform-provider-vcd#699]
### Terraform Version
Terraform v0.12.20
 provider.vcd v3.2.0

### Affected Resource(s)
- vcd_independent_disk

### Original Terraform Configuration file
```hcl
provider "vcd" {
  version = "~> 3.0"
  auth_type = "integrated"
  user = "appsdemo.local"
  password = "xxx"
  org = "APPSDEMOORG"
  url = "https://iaas-vdc.hi.inet/api"
  max_retry_timeout = 5
  allow_unverified_ssl = true
}

data "vcd_catalog_item" "template" {
  catalog = "VDC 4.0 - SIN LICENCIA"
  filter {
    name_regex = "dummy_centos7"
    latest = true
  }
}

resource "vcd_independent_disk" "disk" {
  name = "independent_disk_ejge3"
  vdc = "APPSDEMOORG_PPU_BOE1_01"
  size_in_mb = "3000"
  bus_type = "SCSI"
  bus_sub_type = "VirtualSCSI"
  storage_profile = "GOLD_BOE1_vSAN"
}

resource "vcd_vapp" "vapp" {
  name = "my_tf_vapp3"
  vdc = "APPSDEMOORG_PPU_BOE1_01"
}

resource "vcd_vapp_vm" "vm" {
  vapp_name = vcd_vapp.vapp.name
  name = "my_tf_vm3"
  vdc = vcd_vapp.vapp.vdc
  catalog_name  = data.vcd_catalog_item.template.catalog
  template_name = data.vcd_catalog_item.template.name
  memory = 1024
  cpus = 1
  
  disk {
    name = vcd_independent_disk.disk.name
    bus_number  = 1
    unit_number = 0
  }
}
```

### Second Terraform Configuration file (just "size_in_mb" is changed)
```hcl
provider "vcd" {
  version = "~> 3.0"
  auth_type = "integrated"
  user = "appsdemo.local"
  password = "xxx"
  org = "APPSDEMOORG"
  url = "https://iaas-vdc.hi.inet/api"
  max_retry_timeout = 5
  allow_unverified_ssl = true
}

data "vcd_catalog_item" "template" {
  catalog = "VDC 4.0 - SIN LICENCIA"
  filter {
    name_regex = "dummy_centos7"
    latest = true
  }
}

resource "vcd_independent_disk" "disk" {
  name = "independent_disk_ejge3"
  vdc = "APPSDEMOORG_PPU_BOE1_01"
  size_in_mb = "4000"
  bus_type = "SCSI"
  bus_sub_type = "VirtualSCSI"
  storage_profile = "GOLD_BOE1_vSAN"
}

resource "vcd_vapp" "vapp" {
  name = "my_tf_vapp3"
  vdc = "APPSDEMOORG_PPU_BOE1_01"
}

resource "vcd_vapp_vm" "vm" {
  vapp_name = vcd_vapp.vapp.name
  name = "my_tf_vm3"
  vdc = vcd_vapp.vapp.vdc
  catalog_name  = data.vcd_catalog_item.template.catalog
  template_name = data.vcd_catalog_item.template.name
  memory = 1024
  cpus = 1
  
  disk {
    name = vcd_independent_disk.disk.name
    bus_number  = 1
    unit_number = 0
  }
}
```

### Debug Output
[tf.log and go-vcloud-director.log](https://gist.github.com/ejgarcia/1df8dddd58e28c7fddb9df6a1544a53c)

### Expected Behavior
When an independent disk which is already attached to a VM is requested to be modified, the dependency should be taken into account when destroying the disk. Two possible options:

1. (Ideally) The disk should be detached from the VM, destroyed, recreated, and then re-attached to the VM
2. Alternatively, the VM, as a dependency, would be destroyed, and created again after the disk is resized.

### Actual Behavior
Terraform acknowledges that changing the disk size implies that the disk "must be replaced" (destroyed and recreated), which is OK. However, applying the configuration ends with an error informing that the disk cannot de destroyed because it is attached:

vcd_independent_disk.disk: Destroying... [id=urn:vcloud:disk:009e9765-4777-4334-bdbc-156f780a6e5a]

Error: can not remove disk as it is attached to vm

### Steps to Reproduce
1. Use `terraform apply` with the first file, and finish deployment
2. Replace figure "3000" with "4000" (parameter "size_in_mb" of "vcd_independent_disk")
3. Run `terraform apply` again, and the error occurs when proceeding to execution

### Important Factoids
Tested on vCloud Director 10.0 and 10.2.2

**Comment by szymonbr** [comment]
First of all, I confirm I faced the same issue. But I've got few comments on that.
To resize an independent disk you don't have to destroy it and recreate from scratch. That's only what `vcd_independent_disk` resource tries to do when you increase disk size. However, from vCloud web console you can adjust size of the disk without destroying it and any data loss. The only condition for doing that is to detach the disk from a VM. It can be done online without stopping the VM. Of course, it's better to unmount volumes using the disk priorly.

Nevertheless, the whole process of resizing an independent disk seem to be really painful. To do so you have to:
1. Detach the disk from a VM by terraform
2. Resize the disk  - make manual reconfiguration from vCloud web console / alternatively apply by terraform but data loss occurs because of forcing replacement
3. Reflect disk size change in terraform code and refresh terraform state accordingly if step 2 is done manually.
4. Attach the disk again by terraform

If it can be done semi-manually without any issues in those few steps it should be doable by just applying size change in terraform code as well.

Terraform 1.0.7; vCloud 10.1.3; vcd_provider 3.2.0

**Comment by lvirbalas** [comment]
CC: @vbauzysvmware ^^

**Comment by vbauzysvmware** [comment]
Change is merged. Will be released with 3.6 version.
